src/libraries/System.Net.Sockets/tests/FunctionalTests/CreateSocketTests.cs

```diff
@@ -418,16 +418,17 @@ public async Task Ctor_SafeHandle_Listening_Success(bool shareSafeHandle)
             using var listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
             listener.Bind(new IPEndPoint(IPAddress.Loopback, 0));
             listener.Listen();
-            Assert.Equal(1, listener.GetSocketOption(SocketOptionLevel.Socket, SocketOptionName.AcceptConnection));
 
             using var listenerCopy = new Socket(shareSafeHandle ? listener.SafeHandle : new SafeSocketHandle(listener.Handle, ownsHandle: false));
-            Assert.Equal(1, listenerCopy.GetSocketOption(SocketOptionLevel.Socket, SocketOptionName.AcceptConnection));
+            Assert.False(listenerCopy.Connected);
+            // This will throw if _isListening is set internally. (before reaching any real code)
```

Comment:
I don't understand the comment. Is this saying Connect on the copy will throw but not on the original? Why?

Reply:
It should throw on both. It will not throw InvalidOperationException on copy if we fail to detect that Listen() was called on the handle.